hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,392 @@
+            HyperSpy signal which should be added to the EMD instance.
+        name: string, optional
+            Name of the (used as a key for the `signals` dictionary). If not specified,
+            `signal.metadata.General.title` will be used. If this is an empty string, both name
+            and signal title are set to 'dataset' per default. If specified, `name` overwrites the
+            signal title.
+        metadata: dictionary
+            Dictionary which holds signal specific metadata which will be added to the signal.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        This is the preferred way to add signals to the EMD instance. Directly adding to the
+        `signals` dictionary is possible but does not make sure all metadata are correct. This
+        method is also called in the standard constructor on all entries in the `signals`
+        dictionary!
+
+        """
+        self._log.debug('Calling add_signal')
+        # Create metadata if not present:
+        if metadata is None:
+            metadata = {}
+        # Check and save title:
+        if name is not None:  # Overwrite Signal title!
+            signal.metadata.General.title = name
+        else:
+            if signal.metadata.General.title is not '':  # Take title of Signal!
+                name = signal.metadata.General.title
+            else:  # Take default!
+                name = '__unnamed__'
+                signal.metadata.General.title = name
+        # Save signal metadata:
+        signal.metadata.Signal.add_dictionary(metadata)
+        # Save global metadata:
+        signal.metadata.General.add_node('user')
+        signal.metadata.General.user.add_dictionary(self.user)
+        signal.metadata.General.add_node('microscope')
+        signal.metadata.General.microscope.add_dictionary(self.microscope)
+        signal.metadata.General.add_node('sample')
+        signal.metadata.General.sample.add_dictionary(self.sample)
+        signal.metadata.General.add_node('comments')
+        signal.metadata.General.comments.add_dictionary(self.comments)
+        # Add signal:
+        self.signals[name] = signal
+
+    @classmethod
+    def load_from_emd(cls, filename, load_to_memory=True):
+        """Construct :class:`~.EMD` object from an emd-file.
+
+        Parameters
+        ----------
+        filename : string
+            The name of the emd-file from which to load the signals. Standard format is '*.emd'.
+        load_to_memory: bool, optional
+            If True (default) loads data to memory. If False, enables loading only if requested.
+
+        Returns
+        -------
+        emd: :class:`~.EMD`
+            A :class:`~.EMD` object containing the loaded signals.
+
+        """
+        cls._log.debug('Calling load_from_ems')
+        # Read in file:
+        emd_file = h5py.File(filename, 'r')
+        # Creat empty EMD instance:
+        emd = EMD()

```

Comment:
Bad habits die hard, I did this wrong for a long time and try to break the habit :-)